create-and-save-map/src/main/java/com/esri/arcgismaps/sample/createandsavemap/components/MapViewModel.kt

```diff
@@ -0,0 +1,178 @@
+/* Copyright 2024 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.createandsavemap.components
+
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.viewModelScope
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.AndroidViewModel
+
+import com.arcgismaps.httpcore.authentication.OAuthUserConfiguration
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.Basemap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.ArcGISMapImageLayer
+import com.arcgismaps.mapping.layers.Layer
+import com.arcgismaps.portal.Portal
+import com.arcgismaps.portal.PortalFolder
+import com.arcgismaps.toolkit.authentication.AuthenticatorState
+import com.esri.arcgismaps.sample.createandsavemap.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+
+class MapViewModel(application: Application) : AndroidViewModel(application) {
+
+    // view model to handle popup dialogs
+    val messageDialogVM = MessageDialogViewModel()
```

Comment:
Unless `VM` is a well known abbreviation (e.g. used throughout the samples) I'd spell it out as `messageDialogViewModel`.

Reply:
A quick find-in-files says this is the 14th sample where it's `messageDialogVM`. There aren't any uses of `MessageDialogViewModel` where it's spelt out in full.